### PR TITLE
refactor(util): improve setPanelIDs

### PR DESCRIPTION
**Pull request by Duologic**
Result of today's mobsession.

- Use ~std.mapWithIndex~ std.foldl
- Provide unit test that ensures the order is maintained
- Following comments: ensure collisions do not happen (track index)

### CODE REVIEW

File: grafonnet-base/util/panel.libsonnet [hunk 1]
```diff
@@ -13,22 +13,28 @@ local d = import 'github.com/jsonnet-libs/docsonnet/doc-util/main.libsonnet';
...
+        function(index, panel)
+          panel { id: indexStart + index + init }
+          + (
+            if panel.type == 'row'
+               && 'panels' in panel
+            then {
+              panels:
+                infunc(
+                  panel.panels,
+                  std.length(panels) * (indexStart + index)
```
Comment (joanlopez) [inline grafonnet-base/util/panel.libsonnet:31]: Could this cause collision? 🤔 

For instance, if we have 1 panel plus 2 row panels with 10 panels each:
- The first panel will be identified with `1`
- The panels inside the first row will be identified from `3 * (1 + 1)`: 6, 7, 8, 9, 10... 
- The panels inside the second row will be identified from `3 * (1 + 2)`: 9, 10, 11, 12...

Or am I missing something?

PS: For some reason, I completely missed that during the mob session 🙏🏻

Comment (K-Phoen) [reply grafonnet-base/util/panel.libsonnet:31]: I'm also wondering about collisions :thinking: 

Is there a reason why we don't use a sequential ID for the panels?

Comment (Duologic) [reply grafonnet-base/util/panel.libsonnet:31]: I believe you both are right. This algorithm works for some cases but not all, it definitely comes up when having a lot of panels in a row.

This needs different logic. There is no particaluar reason not to use sequential IDs.

File: grafonnet-base/util/panel.libsonnet [hunk 2]
```diff
@@ -13,22 +13,39 @@ local d = import 'github.com/jsonnet-libs/docsonnet/doc-util/main.libsonnet';
...
+      std.foldl(
+        function(acc, panel)
+          acc {
+            index:  // Track the index to ensure no duplicates exist.
+              acc.index
+              + 1
+              + (if panel.type == 'row'
+                    && 'panels' in panel
+                 then std.length(panel.panels)
+                 else 0),
```
Comment (malcolmholmes) [inline grafonnet-base/util/panel.libsonnet:29]: that's clever

**Review by malcolmholmes — APPROVED** [review]

**Review by K-Phoen — APPROVED** [review]

**Pull request merged**